### DISCUSSION TITLE
Global Application Level Rate Limiting

### DISCUSSION THREAD

**Discussion by addykoder** [nfriedly/express-rate-limit#349] [Q&A] [answered]
I am working on an express app which requires to setup a rate limit globally to the application. Essentially, the request count is not to be based on the user, i.e. different for different users BUT there needs to be a single request count for the whole application. I can't find any useful answer on this over the internet except for the documentation of express-rate-limit has a reference of proxy load balancer which if not configured correctly will make the limit global, but in my application it is configured well and maps the IP of the client. I hope someone can help me out of this

**Comment by gamemaker1** [comment, marked as answer]
Hi @AddyKoder,

You could try to add a rate limiter to the `/` URL, and then subsequently add another rate limiter that rate limits per user. Note that the order in which you apply the limiters is important, make sure the global one is first.

For example,

```ts
import express from 'express'
import { rateLimit } from 'express-rate-limit'

const app = express()

app.use('/', rateLimit({ max: 10000, keyGenerator: () => 'global' })) // overall limiter
app.use('/', rateLimit({ max: 100 })) // per ip limiter
```

Haven't tested this out, but something like this should do the trick.

Regards,
Vedant

**Reply by nfriedly** [reply]
Yeah, this is essentially what I was going to say. The custom `keyGenerator` that returns the same value for every user will turn it into a global rate limiter. I might add a note to the readme about this. (Update: done.)

The order does matter, although I might do the individual one first so that a single individual could not trigger the global rate limit on their own.

**Reply by addykoder** [reply]
Thanks! , that is exactly what I was looking for. The documentation didn't mentioned it clearly before, but now with the relevant information, I think other devs won't face the same problem.

Thanks to the Maintainers @gamemaker1 and @nfriedly for such a quick replay and action.
